File: stars-app/frontend/src/types/plan.ts
export interface PlanSummary {
  contract_id: string;
  plan_name: string;
  state: string;
  enrollment: number;
  py_rating: number;
  projected_rating: number;
  hedis_rating: number;
  cahps_rating: number;
  hos_rating: number;
  partd_rating: number;
  star_gap_to_4: number;
  bonus_eligible: boolean;
}

export interface DomainScore {
  domain: string;
  rating: number;
  on_track_count: number;
  total_count: number;
  critical_count: number;
}

export interface HistoricalRating {
  year: string;
  rating: number;
}

export interface PlanDetail extends PlanSummary {
  historical_ratings: HistoricalRating[];
  domain_scores: DomainScore[];
}

export interface StarSummary {
  total_plans: number;
  total_enrollment: number;
  above_4star_count: number;
  above_4star_pct: number;
  avg_star_rating: number;
  measurement_year: number;
}

File: stars-app/backend/models/plan.py
from pydantic import BaseModel


class PlanSummary(BaseModel):
    contract_id: str
    plan_name: str
    state: str
    enrollment: int
    py_rating: float
    projected_rating: float
    hedis_rating: float
    cahps_rating: float
    hos_rating: float
    partd_rating: float
    star_gap_to_4: float
    bonus_eligible: bool


class DomainScore(BaseModel):
    domain: str
    rating: float
    on_track_count: int
    total_count: int
    critical_count: int


class HistoricalRating(BaseModel):
    year: str
    rating: float


class PlanDetail(PlanSummary):
    historical_ratings: list[HistoricalRating]
    domain_scores: list[DomainScore]


class StarSummary(BaseModel):
    total_plans: int
    total_enrollment: int
    above_4star_count: int
    above_4star_pct: float
    avg_star_rating: float
    measurement_year: int
